gamemodes/terrortown/gamemode/server/sv_roleselection.lua

```diff
@@ -599,7 +609,7 @@ local function UpgradeRoles(plys, subrole, selectableRoles)
 
 	-- now upgrade this role if there are other subroles
 	for sub in pairs(selectableRoles) do
-		if roles.GetByIndex(sub).baserole == subrole then
+		if selectableRoles[sub] >= 1 and roles.GetByIndex(sub).baserole == subrole then
```

Comment:
Empty roles shoudln't be in here

Reply:
This is outdated. Already changed that with the latest update. It's back to 
if roles.GetByIndex(sub).baserole == subrole then
as roles now get removed if they are empty